icu4c/source/i18n/messageformat2_function_registry.cpp

```diff
@@ -19,6 +19,9 @@
 #include "number_types.h"
 #include "uvector.h" // U_ASSERT
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
```

Comment:
Would you mind using the formulation from PR #2250 ?
```suggestion
// The C99 standard suggested that C++ implementations not define PRId64 etc. constants
// unless this macro is defined.
// See the Notes at https://en.cppreference.com/w/cpp/types/integer .
// Similar to defining __STDC_LIMIT_MACROS in unicode/ptypes.h .
#ifndef __STDC_FORMAT_MACROS
#   define __STDC_FORMAT_MACROS
#endif
```

Reply:
@barracuda156 ping?

Reply:
@srl295 Sorry for a delay, will finalize this in a day or two.